src/Avalonia.Base/Collections/AvaloniaDictionary.cs

```diff
@@ -34,6 +34,21 @@ public AvaloniaDictionary(int capacity)
             _inner = new Dictionary<TKey, TValue>(capacity);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvaloniaDictionary{TKey, TValue}"/> class using an IDictionary.
+        /// </summary>
+        public AvaloniaDictionary(IDictionary<TKey, TValue> dictionary, IEqualityComparer<TKey>? comparer = null)
+        {
+            if (dictionary != null)
```

Comment:
Nit: you can remove this check and the `throw` below. Let the `Dictionary`'s constructor throw on its own, as its documented to do so. There's no real point in checking the same thing several times.